pkg/workload/schemachange/operation_generator.go

```diff
@@ -3708,7 +3708,19 @@ func (og *operationGenerator) randType(
 		}
 		return typName, typ, nil
 	}
+
+	pgVectorNotSupported, err := isClusterVersionLessThan(
+		ctx,
+		tx,
+		clusterversion.V24_2.Version())
+	if err != nil {
+		return nil, nil, err
+	}
+
 	typ := randgen.RandSortingType(og.params.rng)
+	for pgVectorNotSupported && typ.Family() == types.PGVectorFamily {
+		typ = randgen.RandSortingType(og.params.rng)
```

Comment:
Out of curiosity: is there something stopping this code from returning pgvector again?

Reply:
yes, the `for` loop will keep iterating until it generates a type that is not `pgvector`